fdbserver/include/fdbserver/DDShardTracker.h

```diff
@@ -24,17 +24,103 @@
 // send request/signal to DDTracker through interface
 // call synchronous method from components outside DDShardTracker
 struct IDDShardTracker {
-	// FIXME: the streams are not used yet
 	Promise<Void> readyToStart;
-	PromiseStream<GetMetricsRequest> getShardMetrics;
-	PromiseStream<GetTopKMetricsRequest> getTopKMetrics;
-	PromiseStream<GetMetricsListRequest> getShardMetricsList;
-	PromiseStream<KeyRange> restartShardTracker;
-
-	// PromiseStream<Promise<int64_t>> averageShardBytes; // FIXME(xwang): change it to a synchronous call
+	FutureStream<GetMetricsRequest> getShardMetrics;
+	FutureStream<GetTopKMetricsRequest> getTopKMetrics;
+	FutureStream<GetMetricsListRequest> getShardMetricsList;
+	FutureStream<Promise<int64_t>> averageShardBytes;
 
 	virtual double getAverageShardBytes() = 0;
 	virtual ~IDDShardTracker() = default;
 };
 
+struct DataDistributionTrackerInitParams {
+	Reference<IDDTxnProcessor> db;
+	UID const& distributorId;
+	Promise<Void> const& readyToStart;
+	PromiseStream<RelocateShard> const& output;
+	Reference<ShardsAffectedByTeamFailure> shardsAffectedByTeamFailure;
+	Reference<PhysicalShardCollection> physicalShardCollection;
+	Reference<AsyncVar<bool>> anyZeroHealthyTeams;
+	KeyRangeMap<ShardTrackedData>* shards = nullptr;
+	bool* trackerCancelled = nullptr;
```

Comment:
`bool*` can be dangerous: say someone means to check `if (*trackerCancelled)`, and instead checks `if (trackerCancelled)`, this won't raise any compilation issues but will have different behavior that's hard to debug. Again, this is probably beyond the scope of this particular refactor. Looks like we use this in many locations.